src/javascript/crypto/e2e/openpgp/pgpmime/mimenode.js

```diff
@@ -130,10 +125,66 @@ e2e.openpgp.pgpmime.MimeNode.prototype.setContent = function(content) {
 
 
 /**
+ * Checks if the proposed boundary is contained within the message.
+ * @param {string} boundary The proposed boundary.
+ * @return {boolean} Returns true if the proposed boundary does not appear
+ *     within the message. Otherwise returns false.
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.isBoundaryValid_ = function(boundary) {
+  // Assign a temporary boundary that will be used in buildMessage().
+  this.boundary_ = 'temporary_boundary';
+  var tempMessage = this.mimeToString_();
+  return !goog.string.contains(tempMessage, boundary);
+};
+
+
+/**
+ * Generates a string that can be used as a MIME message boundary for a node.
+ * @return {string}
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.generateBoundary_ = function() {
+  return '---' + goog.string.getRandomString() +
+      Math.floor(Date.now() / 1000).toString();
+};
+
+
+/**
+ * Sets the MIME message boundary for a node.
+ * Until it succeeds, this function recursively reattempts to generate a unique
+ * boundary for the message (i.e., it attempts to generate a string that doesn't
+ * appear within the original message).
+ * @param {number} bound A bound on the amount of permitted recursions.
+ * @private
+ */
+e2e.openpgp.pgpmime.MimeNode.prototype.setBoundary_ = function(bound) {
+  var boundary = this.generateBoundary_();
```

Comment:
I'll change it to a `do..while` loop here. We're relying on `goog.string.getRandomString()` - do you think that's sufficiently random?